Orange/data/table.py

```diff
@@ -539,225 +590,65 @@ def from_url(cls, url):
-                else:
-                    self.W.resize(old_length)
-            if self.ids.shape[0] == new_length:
-                self.ids.resize(old_length)
-            raise
-
-    def __getitem__(self, key):
-        if isinstance(key, Integral):
-            return RowInstance(self, key)
-        if not isinstance(key, tuple):
-            return self.from_table_rows(self, key)
-
-        if len(key) != 2:
-            raise IndexError("Table indices must be one- or two-dimensional")
-
-        row_idx, col_idx = key
-        if isinstance(row_idx, Integral):
-            if isinstance(col_idx, (str, Integral, Variable)):
-                col_idx = self.domain.index(col_idx)
-                var = self.domain[col_idx]
-                if 0 <= col_idx < len(self.domain.attributes):
-                    return Value(var, self.X[row_idx, col_idx])
-                elif col_idx >= len(self.domain.attributes):
-                    return Value(
-                        var,
-                        self._Y[row_idx,
-                                col_idx - len(self.domain.attributes)])
-                elif col_idx < 0:
-                    return Value(var, self.metas[row_idx, -1 - col_idx])
-            else:
-                row_idx = [row_idx]
-
-        # multiple rows OR single row but multiple columns:
-        # construct a new table
-        attributes, col_indices = self.domain._compute_col_indices(col_idx)
-        if attributes is not None:
-            n_attrs = len(self.domain.attributes)
-            r_attrs = [attributes[i]
-                       for i, col in enumerate(col_indices)
-                       if 0 <= col < n_attrs]
-            r_classes = [attributes[i]
-                         for i, col in enumerate(col_indices)
-                         if col >= n_attrs]
-            r_metas = [attributes[i]
-                       for i, col in enumerate(col_indices) if col < 0]
-            domain = Domain(r_attrs, r_classes, r_metas)
-        else:
-            domain = self.domain
-        return self.from_table(domain, self, row_idx)
+    def add_column(self, col_class, column_types, columns):
+        # TODO: override __setitem__ for strings, autodetermine variable type, default role X
+        """
+        Add one or multiple columns to the table.
+        Column types and column values must be of the same length.
+        If only a single value for a given column exists, the value is used in all rows (broadcasting).
+        :param col_class: Specifies the target column class (either x, y, or meta, see Table.ColumnClass).
+        :param column_types: The column type(s), in the form of Variables.
+        :param columns: The column values. Allows broadcasting single values.
+        """
+        column_types, columns = self._transform_types_columns(column_types, columns)
+        for t, c in zip(column_types, columns):
+            self._column_class_map[t] = Table.ColumnClass.get(col_class)
```

Comment:
I forgot to delete this function when porting to __setitem__. Remnants of previous implementations.